### PR TITLE
feat(e2e): browser+Docker preflight in global-setup (PP-cao)

### PR DESCRIPTION
**Pull request by timothyfroehlich**
## Summary
Adds two fail-fast checks to `e2e/global-setup.ts` before the existing Supabase/Postgres pre-flight:

1. **`checkBrowserBinaries(config)`** — inspects `config.projects` to derive needed browsers (chromium/firefox/webkit), verifies each binary exists via `existsSync(executablePath())`. On miss, throws with `Install with: pnpm exec playwright install --with-deps <names>`. **Only checks browsers the active config actually needs**, so CI's chromium-only install isn't penalized.
2. **`checkDocker()`** — runs `docker info` via `spawnSync` (no shell, fixed argv) with a 5s timeout. On failure: `Start OrbStack: open -a OrbStack`.

Both checks run regardless of `SKIP_SUPABASE_RESET` — that flag only skips DB reset/migration/seed; browsers and the Docker daemon are still required to launch Playwright.

Replaces cryptic mid-run failures like `browserType.launch: Executable doesn't exist` with clear actionable messages before any spec runs.

Also updates the global-setup unit test to pass a mock `FullConfig` (empty projects array disables the browser check for tests focused on DB orchestration) and mocks `spawnSync` for the Docker check.

## Inherited CI failures
`main` is currently red. Inherited failures from main (PP-q9r/PP-e20/PP-jsh/PP-v7g/PP-49m) will appear in CI; not introduced by this PR. See Tim's status comment on PR #1278.

## Test plan
- [x] `pnpm run check` passes (1013 unit tests, including updated global-setup tests)
- [ ] CI Gate green
- [ ] Manual: rename `~/Library/Caches/ms-playwright/chromium-*` and confirm fail-fast with install hint
- [ ] Manual: stop Docker and confirm fail-fast with OrbStack hint

## Related
- Child 3 of epic PP-2on
- Sibling PRs: #1281 (docs), #1282 (worktree tracking), #1283 (Turnstile)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/test/unit/global-setup.test.ts [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-  default: { execSync: execSyncMock },
+  spawnSync: spawnSyncMock,
+  default: { execSync: execSyncMock, spawnSync: spawnSyncMock },
 }));
 
+// The browser-binaries preflight only inspects projects in the config we
+// pass in. An empty `projects` array therefore disables browser checks for
+// these unit tests, keeping coverage focused on the DB orchestration flow.
+const EMPTY_CONFIG = { projects: [] } as unknown as FullConfig;
+
```
[the remark this answers is not in the document]
Comment (timothyfroehlich) [reply src/test/unit/global-setup.test.ts:27]: Added spawnSync mock-call assertions and 4 failure-path scenarios (Docker not running, Docker not installed via ENOENT, missing browser binary, ordering check). —Claude

File: e2e/global-setup.ts [hunk 2]
```diff
@@ -11,17 +14,88 @@ function redactUrl(url: string): string {
...
+  const result = spawnSync("docker", ["info"], {
+    stdio: "ignore",
+    timeout: 5000,
+  });
+  if (result.error || result.status !== 0) {
+    throw new Error(
+      "Docker daemon is not running. Supabase's local stack needs it.\n" +
+        "  Start OrbStack: open -a OrbStack\n" +
+        "  Then wait a few seconds and re-run."
+    );
```
[the remark this answers is not in the document]
Comment (timothyfroehlich) [reply e2e/global-setup.ts:82]: Branched ENOENT (not installed → install hint) vs other errors (daemon down → start hint). —Claude
